src/state/routing/types.ts

```diff
@@ -2,11 +2,20 @@ import { Trade } from '@uniswap/router-sdk'
 import { Currency, Token, TradeType } from '@uniswap/sdk-core'
 
 export enum TradeState {
-  LOADING,
   INVALID,
+  LOADING,
   NO_ROUTE_FOUND,
+  /**
+   * Only valid states should be listed after VALID.
+   * This simplifies validity checks to state >= TradeState.VALID
+   */
   VALID,
   SYNCING,
+  WRAP,
```

Comment:
not sure if this is unhelpful, but `WRAP` doesn't seem like a trade **state** in the same way that these others are.

Reply:
These are strongly coupled (ie a wrap is always an invalid trade), so I'm struggling with how to cleanly express this.
Because of the way the hooks are set up, we'll always execute the trade logic even if it is a wrap.